python/pyspark/mllib/evaluation.py

```diff
@@ -40,16 +40,28 @@ class BinaryClassificationMetrics(JavaModelWrapper):
     >>> metrics.areaUnderPR
     0.83...
     >>> metrics.unpersist()
+    >>> scoreAndLabelsWithOptWeight = sc.parallelize([
+    ...     (0.1, 0.0, 1.0), (0.1, 1.0, 0.4), (0.4, 0.0, 0.2), (0.6, 0.0, 0.6), (0.6, 1.0, 0.9),
+    ...     (0.6, 1.0, 0.5), (0.8, 1.0, 0.7)], 2)
+    >>> metrics = BinaryClassificationMetrics(scoreAndLabelsWithOptWeight)
+    >>> metrics.areaUnderROC
+    0.70...
+    >>> metrics.areaUnderPR
+    0.83...
 
     .. versionadded:: 1.4.0
     """
 
-    def __init__(self, scoreAndLabels):
-        sc = scoreAndLabels.ctx
+    def __init__(self, scoreAndLabelsWithOptWeight):
```

Comment:
Ah, this might be a problem in Python in a way it isn't in Scala. The name of the parameter matters. I think you can no longer call this with `...(scoreAndLabels=...)`. It's surprising if someone would do that, but, possible. I think at the end, maybe we don't change this parameter name to be safe, even though it now also accepts weights. That's not so bad.

And then I wonder, maybe change back the name of the parameter in Scala too? It's really scoreAndLabels(AndWeightsThoughThat'sMoreOfADetail)

Reply:
done, also updated the scala side and the multiclass metrics python and scala classes.  It looks like the python side for regression wasn't updated so I will do that in a separate PR and undo the scala parameter name change there too.